assets/app/scripts/directives/logViewer.js

```diff
@@ -64,6 +64,52 @@ angular.module('openshiftConsole')
               logLinks.scrollTop();
             };
 
+            var bufferedLines = [];
+            // Display all buffered lines.
+            var update = function() {
+              var fragment = document.createDocumentFragment();
+              var logContent = document.getElementById('logContent');
+              angular.forEach(bufferedLines, function(line) {
+                // Manipulate the DOM directly for better performance displaying large log files.
```

Comment:
at least add a comment with the DOM you're building:
```
<div row class="log-line">
  <div class="log-line-number"><div row flex main-axis="end">LINE NUMBER</div></div>
  <div flex class="log-line-text">LOG TEXT</div>
</div>
```
and a todo to swap out for cloneNode